apis/projectcontour/v1alpha1/contourconfig_helpers.go

```diff
@@ -226,16 +224,26 @@ func (e *EnvoyTLS) SanitizedCipherSuites() []string {
 
 func (f FeatureFlags) Validate() error {
 	for _, featureFlag := range f {
-		if _, found := featureFlagsMap[featureFlag]; !found {
+		fields := strings.Split(featureFlag, "=")
+		if _, found := featureFlagsMap[fields[0]]; !found {
 			return fmt.Errorf("invalid contour configuration, unknown feature flag:%s", featureFlag)
 		}
 	}
-
 	return nil
 }
 
 func (f FeatureFlags) IsEndpointSliceEnabled() bool {
-	return slices.Contains(f, featureFlagUseEndpointSlices)
+	// only when the flag: 'useEndpointSlices=false' is exists, return false
```

Comment:
I guess we didnt really consider what we would do when we wanted to change the default for an existing feature flag

another option here if we dont want to change to the `<flag>=<enabled boolean>` would be to:
- by default we assume we are going to use EndpointSlice (feature flags list is empty, or contains `useEndpointSlices`)
- add a new feature flag `useEndpoints`, if that is present then use Endpoints instead
- user cannot provide both flags, it is a validation error if both are specified
- essentially deprecate `useEndpointSlices` flag since it is now the default

I don't know, I'm not super super convinced anymore about the necessity of this feature flags construct tbh, especially if we quickly move to make things default. I guess it can still be useful for things like switching options on the xDS protocol (ADS, Delta, etc.) but we already have a config section for that cc @projectcontour/maintainers @projectcontour/contour-reviewers

Reply:
Sorry for the delay, because I have been on vacation during the Spring Festival for the past two weeks. 
@skriss I refer to the link when make this PR. 
@skriss  @sunjayBhatia TBH, I didn't fully understand and combine your comments, in my impression, the release plan was not divided into alpha and beta,...

Reply:
lets go ahead with what you have there @izturn, it looks like the k8s featuregate mechanism and makes sense to me 👍🏽